src/main/kotlin/domain/card/ShuffledCardDeck.kt

```diff
@@ -0,0 +1,24 @@
+package domain.card
+
+import java.util.Stack
+
+class ShuffledCardDeck(cards: List<Card>) : CardDeck {
+    private val cards = Stack<Card>()
+
+    init {
+        cards.shuffled()
+            .forEach {
+                this.cards.push(it)
+            }
```

Comment:
이렇게도 표현해볼수 있을 것 같아요🙂
```suggestion
        this.cards.addAll(cards.shuffled())
```